### PR TITLE
docs: refactor project configuration docs

### PR DESCRIPTION
**Pull request by mariux**

### CODE REVIEW

File: docs/project-config.md [hunk 1]
```diff
@@ -1,17 +1,18 @@
...
 
-Required version is defined by the attribute `terramate.required_version`
+## The `terramate.required_version` Attribute
+
+Required version is defined by the attribute `terramate.required_version` attribute
 where `required_version` accepts a version constraint string,
-which specifies which versions of Terramate can be inside a Terramate project.
+which specifies which versions of Terramate can be used inside a Terramate project.
+
+We recommend to pin the exact version to use and update the config, when updating Terramate.
```
Comment (soerenmartius) [inline docs/project-config.md:15]:
```suggestion
We recommend pinning the exact version to use and updating the config when updating Terramate.
```

File: docs/project-config.md [hunk 2]
```diff
@@ -20,23 +21,29 @@ Valid examples:
...
+  required_version = ">= 0.1.8, < 0.2.0"
 }
 ```
 
-If no `terramate.required_version` is defined on a project, no versioning
-check will be performed. If one is defined, running `terramate` with a
-incompatible version will result in an error for any Terramate command.
+If no `terramate.required_version` is defined on a project, no versioningss
+check will be performed. If a version constraint is defined, running `terramate` with an
+incompatible version will result in a fatal error.
```
Comment (soerenmartius) [inline docs/project-config.md:38]:
```suggestion
No versioning check will be performed if no `terramate.required_version` is defined on a project. If a version constraint is defined, running `terramate` with an incompatible version will result in a fatal error.
```

File: docs/project-config.md [hunk 3]
```diff
@@ -51,19 +58,18 @@ terramate {
...
 
-The environment in which stacks are going to be executed can be configured
-using the `terramate.config.run.env` block. This block has no labels and
-accepts arbitrary attributes where each attribute represents an environment
-variable that will be evaluated and exported when executing the stack.
-The attributes must always evaluate to strings.
+#### The `terramate.config.run.env` Block
 
-Example:
+In `terramate.config.run.env` block a map of environment variables can be defined that will be set when running command using `terramate run`.
```
Comment (soerenmartius) [inline docs/project-config.md:70]:
```suggestion
In `terramate.config.run.env` block a map of environment variables can be defined that will be set when running a command using `terramate run`.
```

File: docs/project-config.md [hunk 4]
```diff
@@ -95,78 +98,14 @@ terramate {
...
-  }
-}
-```
+In addition Globals (`global.*`) and Metadata (`terramate.*`) are available and are evaluated lazy withing the stack context the commands are executed in.
 
 The `env` namespace is meant to give access to the host environment variables,
 it is read-only, and is only available when evaluating
 `terramate.config.run.env` blocks.
 
 This means that any attributes defined
```
Comment (soerenmartius) [inline docs/project-config.md:107]:
```suggestion
Any attributes defined
```

File: docs/project-config.md [hunk 5]
```diff
@@ -95,78 +98,14 @@ terramate {
...
-  config {
-    run {
-      env {
-        A = "A"
-      }
-    }
-  }
-}
-```
+files, but variable names can NOT be defined twice.
```
Comment (soerenmartius) [inline docs/project-config.md:111]:
```suggestion
files, but variable names can **not** be defined twice.
```

**Review by soerenmartius — CHANGES_REQUESTED** [review]

**Comment by soerenmartius** [comment]
Please install the https://grammarly.com/ extension in your browser for automated spell checks. The credentials for our premium account are in the shared 1password vault.

**Comment by soerenmartius** [comment]
ups, didn't close on purpose (wrong button hit)

File: docs/project-config.md [hunk 6]
```diff
@@ -1,17 +1,18 @@
 # Project Configuration
 
 Terramate does not depend on user configuration and comes with a set of sensible defaults.
-But when defaults doesn't work Terramate allows some project wide configurations
-to be changed.
 
-Configurations that are project wide can be defined only once, and they **MUST**
-be defined at the project root.
+Configurations are valid projec-wide and can be defined only once at the project root.
```
Comment (i4ki) [inline docs/project-config.md:5]:
```suggestion
Configurations are valid project-wide and can be defined only once at the project root.
```

File: docs/project-config.md [hunk 7]
```diff
@@ -95,78 +98,14 @@ terramate {
...
-  config {
-    run {
-      env {
-        TF_PLUGIN_CACHE_DIR = "${terramate.stack.path.absolute}/${global.cache_dir}"
-      }
-    }
-  }
-}
-```
+In addition Globals (`global.*`) and Metadata (`terramate.*`) are available and are evaluated lazy withing the stack context the commands are executed in.
```
[the remark this answers is not in the document]
Comment (i4ki) [reply docs/project-config.md:101]: In the computer theory books, the term is called *lazy evaluation* and *lazy evaluated* is used in past sentences.

**Review by soerenmartius — APPROVED** [review]
lgtm

**Comment by i4ki** [comment]
@mariux needs to pull the main branch here for the "GHA Integration test" be happy.

**Pull request merged**